File: code/mathematical-algorithms/basic/Smallest_digit_in_number/SmallestDigitInNumber.java
/* Part of Cosmos by OpenGenus Foundation */
import java.util.*;
public class SmallestDigitInNumber{
    public static void main(String args[]) {
        int num = 238493;
        System.out.println(smallest_digit(num));
    }

    // returns the smallest integer of a given number.
    public static int smallest_digit(int n)
    {
        int min = n%10;
        n/=10;
        while(n!=0)
        {
            if(min>n%10)
                min=n%10;
            n/=10;
        }
        return min;
    }
}

File: algorithms/mathematical_algorithms/smallest_digit_in_number/smallest_digit_in_number.php
<?php

/**
 * Part of Cosmos by OpenGenus Foundation
 */

/**
 * Finds the smallest digit in number
 *
 * @param int $num number
 * @return int smallest digit
 */
function smallest_digit_in_number(int $num)
{
    return min(array_map('intval', str_split($num)));
}

echo "smallest digit in number\n";
$test_data = [
    [0, 0],
    [9, 9],
    [11, 1],
    [99, 9],
    [19, 1],
    [91, 1],
    [10000, 0],
    [1000090, 0],
];
foreach ($test_data as $test_case) {
    $input = $test_case[0];
    $expected = $test_case[1];
    $result = smallest_digit_in_number($input);
    printf(
        "  input %d, expected %s, got %s: %s\n",
        $input,
        var_export($expected, true),
        var_export($result, true),
        ($expected === $result) ? 'OK' : 'FAIL'
    );
}
